src/wallet/wallet.ts

```diff
@@ -479,12 +620,48 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
     }
 
     if (!this.authToken || !validateJWTExpireDate(this.authToken)) {
-      const sign = this.signMessage(timestampNow);
-      const data = await walletApi.createAuthToken(this, timestampNow, this.xpub!, sign);
-      this.authToken = data.token;
+      let privKey = this.authPrivKey;
+
+      if (!privKey) {
+        // Request the client for the PIN
+        const password = usePassword ? usePassword : await this.requestPassword();
+
+        // Use it to get the words from the storage
+        privKey = wallet.getAuthKey(password);
+      }
+
+      await this.renewToken(privKey, timestampNow);
+    } else {
+      // If we have received the user PIN, we should renew the token anyway
```

Comment:
This renewal comes from this comment from @obiyankenobi:
https://github.com/HathorNetwork/hathor-wallet-mobile/pull/142#discussion_r805417620